sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -103,7 +103,7 @@ object FileFormatWriter extends Logging {
       outputSpec: OutputSpec,
       hadoopConf: Configuration,
       partitionColumns: Seq[Attribute],
-      bucketSpec: Option[BucketSpec],
+      bucketSpecAndHash: Option[(BucketSpec, Boolean)],
```

Comment:
It is hard to know the meaning of this Option parameter. Let's add a `@param` for it?